sdk/keyvault/azure-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/KeyVaultErrorCodeStrings.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.certificates;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Properties;
+
+
+class KeyVaultErrorCodeStrings {
+    static final String ERROR_STRINGS_FILE_NAME = "kvErrorStrings.properties";
+    private static Properties errorStrings;
+    private static HashMap<String, String> errorStringsDefault;
+
+    /**
+     * The property name of Azure Key Vault Credentials required error string.
+     */
+    static final String CREDENTIALS_REQUIRED = "credentials_required";
+
+    /**
+     * The property name of Azure Key Vault Endpoint required error string.
+     */
+    static final String VAULT_END_POINT_REQUIRED = "vault_endpoint_required";
+
+    /**
+     *  Gets the error String for the specified property.
+     *
+     * @param propertyName the property name for which error string is required.
+     * @return The {@link String value} containing the error message.
+     */
+    static String getErrorString(String propertyName) {
+        loadProperties();
+        return errorStrings.getProperty(propertyName);
+    }
+
+    private static void loadProperties() {
+        if (errorStrings != null) {
+            return;
+        }
+        try (InputStream fileInputStream = KeyVaultErrorCodeStrings.class.getClassLoader().getResource((ERROR_STRINGS_FILE_NAME)).openStream()) {
+            errorStrings = new Properties();
+            errorStrings.load(fileInputStream);
+        } catch (IOException ex) {
+            ex.printStackTrace();
```

Comment:
throw exception instead of using printStackTrace()

Reply:
This is a tricky case, 
There is no backup mechanism for error strings.
If this throws exception it will bubble up to places like public classes consturctors and will need to be handled there with a backup error string.

Its better to formalize the approach of using backup error strings with @jonathangiles

Currently adding a log message here for IO exception and handling the exception here itself for now.